config/reframe_config_template.j2

```diff
@@ -0,0 +1,90 @@
+                    # Resources for testing this partition (https://reframe-hpc.readthedocs.io/en/stable/config_reference.html#config.systems.partitions.resources)
+                    'resources' : [{'name': 'switches',
+                                    'options': ['--switches={num_switches}']},
+                                   {'name': 'gres',
+                                   'options': ['--gres={gres}']},
+                                   {'name': 'memory',
+                                   'options': ['--mem={mem_per_node}']}],
+                    # https://reframe-hpc.readthedocs.io/en/stable/config_reference.html#config.systems.partitions.extras
+                    'extras' : {{partition.extras}},
+                    # https://reframe-hpc.readthedocs.io/en/stable/config_reference.html#config.systems.partitions.env_vars
+                    'env_vars' : {{partition.env_vars}},
+                    {% if partition.devices %}
+                    # Check if any specific devices were found in this node type
+                    # The gpus found in slurm GRes will be specified here
+                    'devices' : [
+                    {% for dev in partition.devices %}
+                        { 'type' : '{{dev.type}}',
+                          'num_devices' : {{dev.num_devices}}
+                                },
+                    {% endfor %}
+                    ],
+                    {% endif %}
+                    {% if partition.container_platforms %}
+                    # Check if any container platforms are available in these nodes and add them
+                    # https://reframe-hpc.readthedocs.io/en/stable/config_reference.html#container-platform-configuration
+                    'container_platforms' : [
+                    {% for c_p in partition.container_platforms %}
+                        { 'type' : '{{c_p.type}}', # Type of container platform 
+                        {% if c_p.modules %}
+                        # Specify here the modules required to run the container platforms (if any)
+                          'modules' : {{c_p.modules}}
+                        {% endif %}
+                        },
+                    {% endfor %}
+                    ],
+                    {% endif %}
+                    {% endif %}
+                    {% if partition.access %}
+                    # Options passed to the job scheduler in order to submit a job to the specific nodes in this partition
+                    'access' : {{partition.access}},
+                    {% endif %}
+                    {% if partition.features %}
+                    # Node features detected in slurm
+                    'features' : {{partition.features}},
+                    {% endif %}
+                },
+                {% endfor %}
+            ],
+        },
+    ],
+    # The environments cannot be automatically detected, check the following links for reference
+    'environments': {{environments}}
```

Comment:
Since we cannot automatically detect envs, I would prefer to have an empty list and the links that you have now as python comments. Because ideally the automatic config should be able to work out of the box